src/core/Ros.js

```diff
@@ -44,6 +44,7 @@ function Ros(options) {
 
   // begin by checking if a URL was given
   if (options.url) {
+    this.socketio = (options.url.indexOf('http') !== -1);
```

Comment:
This is a fairly fragile way to check if socket.io is requested. It'd be better to use define a new option "transportLibrary" defaulting to "websocket" and accepting "socket.io" (someone might have interest in adding sockjs).